### PR TITLE
adding 'set-exit-code' flag

### PR DESCRIPTION
**Pull request by benzaita**
Defaults to false. When true, sets the exit-code to 1 if any tests
failed. This is useful for automation.

### CODE REVIEW

File: parser.go [hunk 1]
```diff
@@ -148,3 +148,17 @@ func findTest(tests []*Test, name string) *Test {
 	}
 	return nil
 }
+
+func (r *Report) Failures() (count int) {
```
Comment (jstemmer) [inline parser.go]: Can you add a comment to this function? Also, the named parameter is not needed here, could you remove it as well as the naked return on line 163?

File: go-junit-report.go [hunk 2]
```diff
@@ -32,4 +34,12 @@ func main() {
...
 		os.Exit(1)
 	}
+
+	if setExitCode {
+		if report.Failures() > 0 {
+			os.Exit(1)
+		}
+	}
+
+	os.Exit(0)
```
Comment (jstemmer) [inline go-junit-report.go]: The exit code is 0 by default, this line can be removed.

**Comment by jstemmer** [comment]
Looks good. If you fix the few minor comments I added and squash your commits then I can merge it.

**Comment by benzaita** [comment]
Thanks. I will do that as soon as I can.

**Comment by jstemmer** [comment]
Thanks for contributing!

**Pull request merged**